src/extension_utils/utils.ts

```diff
@@ -317,53 +353,63 @@ export const promptInstallPythonDependencies = (
                         } else {
-                            return false;
+                            return "";
                         }
                     });
             }
         });
 };
-export const getTelemetryState = () => {
-    return vscode.workspace
-        .getConfiguration()
-        .get("telemetry.enableTelemetry", true);
+
+export const getPythonVenv = async (context: vscode.ExtensionContext) => {
+    const subFolder = os.platform() === "win32" ? "Scripts" : "bin";
+
+    return getPathToScript(
+        context,
+        path.join(CONSTANTS.FILESYSTEM.PYTHON_VENV_DIR, subFolder),
+        HELPER_FILES.PYTHON_EXE
+    );
 };
 
-export const installPythonDependencies = async (
+export const installPythonVenv = async (
     context: vscode.ExtensionContext,
-    pythonExecutable: string,
-    pathToLibs: string
+    pythonExecutable: string
 ) => {
-    let installed: boolean = false;
-    try {
-        vscode.window.showInformationMessage(
-            CONSTANTS.INFO.INSTALLING_PYTHON_DEPENDENCIES
-        );
+    const pathToEnv: string = getPathToScript(
+        context,
+        CONSTANTS.FILESYSTEM.PYTHON_VENV_DIR
+    );
 
-        const requirementsPath: string = getPathToScript(
-            context,
-            CONSTANTS.FILESYSTEM.OUTPUT_DIRECTORY,
-            "requirements.txt"
-        );
+    vscode.window.showInformationMessage(CONSTANTS.INFO.INSTALLING_PYTHON_VENV);
 
-        // run command to download dependencies to out/python_libs
-        const { stdout } = await exec(
-            `${pythonExecutable} -m pip install -r ${requirementsPath} -t ${pathToLibs}`
-        );
-        console.info(stdout);
-        installed = true;
+    const pythonPath: string = await getPythonVenv(context);
 
-        vscode.window.showInformationMessage(CONSTANTS.INFO.SUCCESSFUL_INSTALL);
+    try {
+        // make venv
+        // run command to download dependencies to out/python_libs
+        await executePythonCommand(pythonExecutable, `-m venv "${pathToEnv}"`);
     } catch (err) {
         vscode.window
             .showErrorMessage(
-                CONSTANTS.ERROR.DEPENDENCY_DOWNLOAD_ERROR,
+                `Virtual environment for download could not be completed. Using original interpreter at: ${pythonExecutable}.`,
+                DialogResponses.READ_INSTALL_MD
+            )
+            .then((selection: vscode.MessageItem | undefined) => {
+                if (selection === DialogResponses.READ_INSTALL_MD) {
+                    open(CONSTANTS.LINKS.INSTALL);
+                }
+            });
+
+        console.error(err);
+
+        return pythonExecutable;
+    }
+
+    if (!(await installDependencies(context, pythonPath))) {
+        vscode.window
+            .showErrorMessage(
+                `Dependency download for the custom virtual environment for download could not be completed. Using original interpreter at: ${pythonExecutable}.`,
```

Comment:
Should we use a constant error string instead for these error messages?

Reply:
Oh right! Thanks for catching that :)

Reply:
Since the prompt includes a variable name, should I make part of the string a constant? Or should I leave it be?

Reply:
We should be able to declare it as a constant with a paremeter for the variable name ("%1", "%s", or something of the sorts)?

Reply:
@xnkevinnguyen I'm redirecting this question to you! :'''D is this possible?

Reply:
I think it's ok to do this for template strings. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals, if you want to do constant, you can make a function that returns the value, but showing error message here should be ok as long as it's consistently done this way

Reply:
Ugh, I couldn't find anything super easy <_< Similar to what Kevin suggested, it should be all right to declare a constant as the string is?
`'Dependency download for the custom virtual environment for download could not be completed. Using original interpreter at: ${pythonExecutable}.'`

Reply:
You could also create a wrapper, as it's likely we will use more of that in the future?
https://stackoverflow.com/a/7790858

Reply:
@isadorasophia is that ok? ^

Reply:
it can be added as a refactoring task in the future if it's needed! It may too much overhead for this PR :(((((

Reply:
No worries! You can just create a work item for that (mostly so we don't forget haha) and deal with it afterwards